apps/vscode-extension/src/commands/extract/styling/extract-all-to-variable.command.ts

```diff
@@ -0,0 +1,59 @@
+import { EOL } from 'node:os';
+import { ExtensionContext, Position, Range, TextEditor, window } from 'vscode';
+import { generateVariableName, insertVariable, isPropertyValue } from './helpers';
+
+/**
+ * Replace all the interested lines with an Otter variable
+ *
+ * @param _context
+ */
+export function extractAllToVariable(_context: ExtensionContext) {
+  return async (editor: TextEditor) => {
+    const isPropertyLineRegExp = /^ *[^ .$#]+ *: *([^$]+) *;/;
```

Comment:
as discussed, you might detect css pseudo class such as :not or :first-child etc.

Reply:
Added a rule to exclude multiple `:`.
This combined with the final `;` should do the job